lua/ulx_yeet/sh_yeet.lua

```diff
@@ -0,0 +1,176 @@
+if SERVER then
+    ULib.ucl.registerAccess( "ulx physgunragdollplayer", ULib.ACCESS_ADMIN, "Ability to physgun ragdoll other players.", "Other" )
+end
+
+CreateConVar( "ulx_physgun_ragdoll_velocity", 75, { FCVAR_REPLICATED, FCVAR_ARCHIVE }, "The velocity required for a physgunned player to turn into a ragdoll on release.", 0 )
+
+local function savePlayer( ply )
+    local result = {}
+
+    result.health = ply:Health()
+    result.armor = ply:Armor()
+
+    if ply:GetActiveWeapon():IsValid() then
+        result.currentWeapon = ply:GetActiveWeapon():GetClass()
+    end
+
+    local weapons = ply:GetWeapons()
+
+    for _, weapon in ipairs( weapons ) do
+        result.weapondata = {}
+        printname = weapon:GetClass()
+        result.weapondata[ printname ] = {}
+        result.weapondata[ printname ].clip1 = weapon:Clip1()
+        result.weapondata[ printname ].clip2 = weapon:Clip2()
+        result.weapondata[ printname ].ammo1 = ply:GetAmmoCount( weapon:GetPrimaryAmmoType() )
+        result.weapondata[ printname ].ammo2 = ply:GetAmmoCount( weapon:GetSecondaryAmmoType() )
+    end
+    ply.cfcYeetData = result
+end
+
+local function restorePlayer( ply )
+    local data = ply.cfcYeetData
+    ply:SetParent()
+    ply:SetHealth( data.health )
+    ply:SetArmor( data.armor )
+
+    for weaponClass, infoTable in pairs( data.weapondata ) do
+        ply:Give( weaponClass )
```

Comment:
ply:Give() returns the weapon given to the player. You could possibly use that instead of ply:GetWeapon

if theres a chance the player already has the weapon you might need to do a NULL check because "It will return NULL if the player already has the weapon".

Reply:
This basically already covers that senario